### ISSUE TITLE
UpdateSentinelAddressList is failing with error: EndPoints must be unique

### ISSUE THREAD

**Issue by ejsmith** [StackExchange/StackExchange.Redis#1430]
I am getting the following error attempting to use sentinel support. I'm guessing this is some sort of concurrency issue, but I'm not sure. I will try and get a repro together.

```
EndPoints must be unique (Parameter 'item')
System.ArgumentException: EndPoints must be unique (Parameter 'item')
   at StackExchange.Redis.EndPointCollection.InsertItem(Int32 index, EndPoint item) in /_/src/StackExchange.Redis/EndPointCollection.cs:line 71
   at StackExchange.Redis.ConnectionMultiplexer.UpdateSentinelAddressList(String serviceName) in /_/src/StackExchange.Redis/ConnectionMultiplexer.cs:line 2400
   at StackExchange.Redis.ConnectionMultiplexer.SwitchMaster(EndPoint switchBlame, ConnectionMultiplexer connection, TextWriter log) in /_/src/StackExchange.Redis/ConnectionMultiplexer.cs:line 2376
   at StackExchange.Redis.ConnectionMultiplexer.GetSentinelMasterConnection(ConfigurationOptions config, TextWriter log) in /_/src/StackExchange.Redis/ConnectionMultiplexer.cs:line 2255
   at Exceptionless.Insulation.Bootstrapper.GetRedisConnection(Dictionary`2 options) in /app/src/Exceptionless.Insulation/Bootstrapper.cs:line 166
```

**Comment by ejsmith** [comment]
Is there any reason why `EndPointCollection.InsertItem` wouldn't just return if the item is already in the collection? I don't really understand why you would want to blow up. Seems like there is no harm in basically assuring that an item is in the collection.

**Comment by ejsmith** [comment]
I guess maybe just because you are expecting to insert it at a specific index and if it already exists in another index position then it can not complete the requested operation.

**Comment by ejsmith** [comment]
I added a `TryAdd` method to the `EndPointCollection` class that just adds the endpoint if it doesn't already exist.

**Comment by NickCraver** [comment]
This was fixed in #1431, closing out to cleanup!
